ql/src/java/org/apache/hadoop/hive/ql/exec/CommonMergeJoinOperator.java

```diff
@@ -240,6 +249,17 @@ public void process(Object row, int tag) throws HiveException {
 
     byte alias = (byte) tag;
     List<Object> value = getFilteredValue(alias, row);
+
+    if (shortcutExtensionRows && isOuterJoinExtensionRow(tag, value)) {
+      int type = condn[0].getType();
+      if (tag == 0 && (type == JoinDesc.LEFT_OUTER_JOIN || type == JoinDesc.FULL_OUTER_JOIN)) {
+        emitExtensionRow(tag, value);
```

Comment:
I applied this patch and added a log message to emitExtensionRow() and ran queries with full , left and right outer join (with the extra single table predicate in ON clause) but only see the log msg for full outer join, not for left/right. It is unclear which situations would lead to extension rows for left/right outer join.

Reply:
it have stopped for me for other outer joins ; however I had to disable auto.join.converion to have that - without that the join was converted into a mapjoin
I will make some further tests for different types of joins to see if we don't have any issues